stepactions/eaas-create-ephemeral-cluster-hypershift-aws/0.1/eaas-create-ephemeral-cluster-hypershift-aws.yaml

```diff
@@ -51,40 +57,45 @@ spec:
       value: "$(params.insecureSkipTLSVerify)"
     - name: TIMEOUT
       value: "$(params.timeout)"
+    - name: IMAGE_CONTENT_SOURCES
+      value: "$(params.imageContentSources)"
   script: |
     #!/bin/bash
     set -eo pipefail
 
     cat <<EOF > cti.yaml
+    ---
     apiVersion: clustertemplate.openshift.io/v1alpha1
     kind: ClusterTemplateInstance
     metadata:
       generateName: cluster-
     spec:
       clusterTemplateRef: hypershift-aws-cluster
-      parameters:
-        - name: instanceType
-          value: $INSTANCE_TYPE
-        - name: version
-          value: $VERSION
-        - name: timeout
-          value: $TIMEOUT
+      parameters: []
     EOF
 
+    yq -i '.spec.parameters += {"name": "instanceType", "value": strenv(INSTANCE_TYPE)}' cti.yaml
+    yq -i '.spec.parameters += {"name": "version", "value": strenv(VERSION)}' cti.yaml
+    yq -i '.spec.parameters += {"name": "timeout", "value": strenv(TIMEOUT)}' cti.yaml
+    yq -i '.spec.parameters += {"name": "imageContentSources", "value": strenv(IMAGE_CONTENT_SOURCES)}' cti.yaml
```

Comment:
Just to double-check: the `value` should be a yaml string, not the object parsed from that string? (I think you would want `env()` rather than `strenv()` for the latter)

Reply:
Yes, the `value` should always be parsed as a string. The yq [docs](https://mikefarah.gitbook.io/yq/operators/env-variable-operators) indicate `strenv()` should be used over `env()` and that was my experience when testing with both.

```
env which takes a single environment variable name and parse the variable as a yaml node (be it a map, array, string, number of boolean)

strenv which also takes a single environment variable name, and always parses the variable as a string.
```